### PR TITLE
Use flutter.js bootstrapping

### PR DESCRIPTION
**Pull request by johnpryan**
This changes sketch_pad's bootstrapping to use flutter.js.

Fixes https://github.com/dart-lang/dart-pad/issues/2261

### CODE REVIEW

**Comment by johnpryan** [comment]
This will depend on https://github.com/flutter/engine/pull/49782/files getting merged first, so leaving this as a draft for now.

**Comment by johnpryan** [comment]
cc: @ditman

**Comment by johnpryan** [comment]
This changes some dart_services code, so you need to test by running dart_services (`grind build-storage-artifacts` and `dart bin/server.dart` and use`?channel=localhost` in the URL)

**Comment by johnpryan** [comment]
@devoncarew I've tested this on the latest `stable` channel and everything is working smoothly. PTAL when you can.

File: pkgs/sketch_pad/web/frame.js
Comment (ditman) [inline pkgs/sketch_pad/web/frame.js:38]:
```suggestion
    onEntrypointLoaded: async function(engineInitializer) {
      let appRunner = await engineInitializer.initializeEngine();
      appRunner.runApp();
    }
```

You can write this with `async/await` in JS and it turns out a little bit prettier IMO.

**Review by ditman — APPROVED** [review]
There's a new, new flutter.js initialization coming, but that's not documented yet. This LGTM!

**Review by devoncarew — APPROVED** [review]
rslgtm for my part, though I'd rely on @ditman for the bulk of the review

**Comment by parlough** [comment]
In general, looks great to me, but due to the dart-services changes is there a desync with clients that could cause them to not work temporarily if connecting to an older server? Same for my temporary use of the legacy frontend.

**Comment by johnpryan** [comment]
@parlough you're right, let's hold off until we've full migrated the websites.

**Comment by johnpryan** [comment]
@parlough have we migrated the websites?

**Review by parlough — APPROVED** [review]
The websites are both using SketchPad now so feel free to land this when you're ready. Thanks for working on this and I appreciate the patience :)

**Comment by johnpryan** [comment]
This is working for me after merging with `main`, so I'm going to land this and keep testing.

File: pkgs/dartpad_ui/web/frame.js
Comment (kevmoo) [inline pkgs/dartpad_ui/web/frame.js:34]: I *think* you can even drop this bit. This is wiring up the default behavior I believe.

Right @ditman  ?

**Comment by johnpryan** [comment]
This is causing a CORS error, so I'm going to roll this back. https://github.com/dart-lang/dart-pad/pull/2959

**Comment by johnpryan** [comment]
```
Access to fetch at 'https://dartpad.dev/canvaskit/chromium/canvaskit.wasm' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Pull request merged**
